### DISCUSSION TITLE
Augustus package in the Debian (+ Ubuntu, etc.) official repositories

### DISCUSSION THREAD

**Discussion by vv221** [Keriew/augustus#1147] [General]
I worked recently on getting julius into the official Debian repositories, cf. https://github.com/bvschaik/julius/issues/739

Once it has been available for a couple weeks/months, so most mistakes I made in the initial packaging (I don’t doubt there are some) are spotted and fixed, I would like to use it as a basis to package augustus too. The goal is to make it readily available to all Debian users (and users of derivatives like Ubuntu or Linux Mint), so it would only be one `apt install augustus` away.

They would still need to use some tools like [game-data-packager] or [./play.it] to package the non-redistributable game data, but I think having the engine available from the distribution repositories would make these tools much easier to use.

[game-data-packager]: https://wiki.debian.org/Games/GameDataPackager
[./play.it]: https://wiki.debian.org/Games/PlayIt

If you are OK with that, I plan to use this here discussion as a way to share any information about the activity on the Debian front. Keeping in mind that it might still be a couple weeks, or even months, before I start any actual work on this new package.

**Comment by crudelios** [comment]
Here I am again! 🙂 

I'm glad you're interested in creating an augustus deb package as well.

The process should be similar to julius with three main differences:

1. I think it was in augustus, and not julius, where I trimmed tinyfiledialogs a lot even for linux, so I'm not sure tkinter is useful as a dependency here - I'll check.

2. Augustus runs on any version of SDL2, but SDL2 >= 2.0.24 is recommended due to it using hardware rendering.

3. Augustus provides extra assets which need to be distributed along with the package.

    Those assets need to be installed in an `assets` subdirectory in one of the following locations (the search is performed in the order below and stops on the first match it finds):

    - `~/.local/share/augustus-game`
    - `/usr/share/augustus-game`
    - `/usr/local/share/augustus-game`
    - `/opt/augustus-game`

    If a different directory needs to be used let me know and I'll add it.

**Reply by vv221** [reply]
> Augustus runs on any version of SDL2, but SDL2 >= 2.0.24 is recommended due to it using hardware rendering.

Debian unstable (the target for new packages) currently provides 2.30.9, so we can expect that one (or newer) to be available.

> Augustus provides extra assets which need to be distributed along with the package.

This part is going to be a bit tricky, but I know it can be handled by Debian source packages. And it will give me an opportunity to learn how to handle that ;)

On Debian games-related stuff tends to be installed in /usr/share/games, not directly in /usr/share, so support for `/usr/share/games/augustus-game` would be welcome. We could patch it ourselves, but it would be even better if you could add it upstream.

---

By the way, this time the name "augustus" is already claimed by some software about genomic sequences. I think I will use "augustus-game" to mirror the choice you made on your side.

**Comment by crudelios** [comment]
> On Debian games-related stuff tends to be installed in /usr/share/games, not directly in /usr/share, so support for /usr/share/games/augustus-game would be welcome. We could patch it ourselves, but it would be even better if you could add it upstream.

I also agree native support is the way to go. I'll add that directory ti the list.

> By the way, this time the name "augustus" is already claimed by some software about genomic sequences. I think I will use "augustus-game" to mirror the choice you made on your side.

Agreed.
